src/app/modules/spcp/spcp.controller.ts

```diff
@@ -117,3 +117,44 @@ export const addSpcpSessionInfo: RequestHandler<ParamsDictionary> = async (
       return next()
     })
 }
+
+/**
+ * Checks if user is SPCP-authenticated before allowing submission
+ * @param req - Express request object
+ * @param res - Express response object
+ * @param next - Express next middleware function
+ */
+export const isSpcpAuthenticated: RequestHandler<ParamsDictionary> = (
+  req,
+  res,
+  next,
+) => {
+  const { authType } = (req as WithForm<typeof req>).form
```

Comment:
I see we have multiple identical `WithForm` types declared in the codebase - can we define it in a single place?

Reply:
yeah true, I think we've been lax about it since those types are eventually going to disappear but it's probably good to de-duplicate them anyway

Reply:
will address this in a separate PR!